### PR TITLE
couple of small refactors

### PR DESCRIPTION
**Pull request by danieleades**

### CODE REVIEW

**Comment by davegarred** [comment]
Could you resolve these conflicts?

File: tests/lib.rs [hunk 1]
```diff
@@ -251,7 +251,7 @@ fn test_framework_test() {
         .when(TestCommand::ConfirmTest(ConfirmTest {
             test_name: test_name.to_string(),
         }))
-        .then_expect_error_message("test already performed")
+        .then_expect_error_message("test already performed");
```
Comment (davegarred) [inline tests/lib.rs:254]: Why remove the semicolons?

File: src/persist/replay.rs [hunk 2]
```diff
@@ -123,7 +123,7 @@ mod test {
         async fn dispatch(&self, _aggregate_id: &str, events: &[EventEnvelope<MyAggregate>]) {
             let mut event_list = self.events.lock().unwrap();
             for event in events {
-                event_list.push(event.to_owned());
+                event_list.push(event.clone());
```
Comment (davegarred) [inline src/persist/replay.rs:126]: I think `to_owned` is more clear here.

**Review by davegarred — CHANGES_REQUESTED** [review]

File: src/persist/upcaster.rs [hunk 3]
```diff
@@ -62,18 +62,10 @@ impl FromStr for SemanticVersion {
...
-            u32::from_str(split_version.get(2).unwrap())?
-        };
-        let minor_version = if split_version.len() < 2 {
-            0
-        } else {
-            u32::from_str(split_version.get(1).unwrap())?
-        };
-        let major_version = u32::from_str(split_version.first().unwrap())?;
+        let mut split_version = event_version.split('.').fuse();
+        let major_version = u32::from_str(split_version.next().unwrap())?;
```
Comment (davegarred) [inline src/persist/upcaster.rs:66]: You've removed the length check so we can't safely `unwrap` here.

**Review by davegarred — CHANGES_REQUESTED** [review]

**Comment by davegarred** [comment]
Commits should be squashed.

File: tests/lib.rs (hunk 1 above)
Comment (danieleades) [reply tests/lib.rs:254]: i didn't, i added a semicolon.
semicolons are optional if the the expression returns the unit value, but i think it's better to use them consistently (and indeed there's a pedantic clippy lint for this)

File: src/persist/upcaster.rs (hunk 3 above)
Comment (danieleades) [reply src/persist/upcaster.rs:66]: no, i think this is ok- even if the 'event_version' is an empty string, `event_version.split('.').next()` will evaluate to `Some("")`, so the unwrap is sound.

You can see that from one of the examples in the docs for the `split` method-

```rust
let v: Vec<&str> = "".split('X').collect();
assert_eq!(v, [""]);
```

Of course if the string is empty, `u32::from_str` will fail

**Comment by danieleades** [comment]
> Commits should be squashed.

sure. Are 'squash merges' enabled for this repo?

File: src/persist/replay.rs (hunk 2 above)
Comment (danieleades) [reply src/persist/replay.rs:126]: sure

File: tests/lib.rs (hunk 1 above)
Comment (davegarred) [reply tests/lib.rs:254]: Derp. Thanks!

**Comment by davegarred** [comment]
> > Commits should be squashed.
> 
> sure. Are 'squash merges' enabled for this repo?

They are available but human-made messages are preferable (and easier to review).

**Comment by danieleades** [comment]
> > > Commits should be squashed.
> > 
> > 
> > sure. Are 'squash merges' enabled for this repo?
> 
> They are available but human-made messages are preferable (and easier to review).

i mean in the sense that you can create a PR with multiple commits, and then just squash them all down with a human-generated commit message when you merge.

Sometimes that's easier to review, since the edit history over time during process is preserved.

I'm happy either way!

File: src/persist/upcaster.rs (hunk 3 above)
Comment (davegarred) [reply src/persist/upcaster.rs:66]: Very good, thank you.

**Review by davegarred — APPROVED** [review]

**Pull request merged**